docs/apidiffs/current_vs_latest/opentelemetry-sdk-trace.txt

```diff
@@ -1,2 +1,8 @@
 Comparing source compatibility of opentelemetry-sdk-trace-1.44.0-SNAPSHOT.jar against opentelemetry-sdk-trace-1.43.0.jar
-No changes.
\ No newline at end of file
++++  NEW INTERFACE: PUBLIC(+) ABSTRACT(+) io.opentelemetry.sdk.trace.data.ExceptionEventData  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW INTERFACE: io.opentelemetry.sdk.trace.data.EventData
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW METHOD: PUBLIC(+) STATIC(+) io.opentelemetry.sdk.trace.data.ExceptionEventData create(io.opentelemetry.sdk.trace.SpanLimits, long, java.lang.Throwable, io.opentelemetry.api.common.Attributes)
```

Comment:
@jkwatson its odd for the create method to accept `SpanLimits` as an argument. Normally the application of limits is handled in SDK internals. We can see this pattern play out in the `EventData.create` contract:
```
static EventData create(
      long epochNanos, String name, Attributes attributes, int totalAttributeCount)
```
The attributes are provided with an additional `totalAttributeCount` indicating how many attributes were originally present.

The reason we do it differently for `ExceptionEventData` appears to be because `ImmutableExceptionEvent#getAttributes()` doesn't compute the attributes until it is called (and then it memoizes them), preventing additional allocations on the hot path.

Not sure there's a way to avoid `SpanLimits` being passed as an argument if we stick with that argument.